eth/core/chain.go

```diff
@@ -135,3 +136,11 @@ func NewChain(host PolarisHostChain) *blockchain { //nolint:revive // only used
 func (bc *blockchain) Config() *params.ChainConfig {
 	return bc.cp.ChainConfig()
 }
+
+// SetGenesisBlock sets the value of currentBlock and finalizedBlock
+// to the genesisBlock on chain genesis.
+func (bc *blockchain) SetGenesisBlock(genesisBlock *types.Block) error {
+	bc.currentBlock.Store(genesisBlock)
```

Comment:
don't do this, create a new thing in the blockchain called `genesisBlock`

Reply:
we also do need the current block and finalized block to point to genesisBlock before the first block is processed

Reply:
oh this is true i guess